body-tracking-samples/sample_unity_bodytracking/Assets/Scripts/TrackerHandler.cs

```diff
@@ -0,0 +1,130 @@
+        parentJointMap[JointId.SpineNavel] = 0;
+        parentJointMap[JointId.SpineChest] = 1;
+        parentJointMap[JointId.Neck] = 2;
+        parentJointMap[JointId.ClavicleLeft] = 2;
+        parentJointMap[JointId.ShoulderLeft] = 4;
+        parentJointMap[JointId.ElbowLeft] = 5;
+        parentJointMap[JointId.WristLeft] = 6;
+        parentJointMap[JointId.HandLeft] = 7;
+        parentJointMap[JointId.HandTipLeft] = 8;
+        parentJointMap[JointId.ThumbLeft] = 8;
+        parentJointMap[JointId.ClavicleRight] = 2;
+        parentJointMap[JointId.ShoulderRight] = 11;
+        parentJointMap[JointId.ElbowRight] = 12;
+        parentJointMap[JointId.WristRight] = 13;
+        parentJointMap[JointId.HandRight] = 14;
+        parentJointMap[JointId.HandTipRight] = 15;
+        parentJointMap[JointId.ThumbRight] = 15;
+        parentJointMap[JointId.HipLeft] = 1;
+        parentJointMap[JointId.KneeLeft] = 18;
+        parentJointMap[JointId.AnkleLeft] = 19;
+        parentJointMap[JointId.FootLeft] = 20;
+        parentJointMap[JointId.HipRight] = 1;
+        parentJointMap[JointId.KneeRight] = 22;
+        parentJointMap[JointId.AnkleRight] = 23;
+        parentJointMap[JointId.FootRight] = 24;
+        parentJointMap[JointId.Head] = 0;
+        parentJointMap[JointId.Nose] = -1;
+        parentJointMap[JointId.EyeLeft] = -1;
+        parentJointMap[JointId.EarLeft] = -1;
+        parentJointMap[JointId.EyeRight] = -1;
+        parentJointMap[JointId.EarRight] = -1;
+    }
+
+    public void updateTracker(BackgroundData trackerFrameData)
+    {
+        //this is an array in case you want to get the n closest bodies
+        int[] closestBodies = new int[1] { 0 };
+        findClosestTrackedBody(trackerFrameData, closestBodies);
+
+        // render the closest body
+        Body skeleton = trackerFrameData.Bodies[closestBodies[0]];
+        renderSkeleton(skeleton, 0);
+    }
+
+    int findIndexFromId(BackgroundData frameData, int id)
+    {
+        int retIndex = -1;
+        for (int i = 0; i < (int)frameData.NumOfBodies; i++)
+        {
+            if ((int)frameData.Bodies[i].Id == id)
+            {
+                retIndex = i;
+                break;
+            }
+        }
+        return retIndex;
+    }
+
+    private void findClosestTrackedBody(BackgroundData trackerFrameData, int[] bodies)
+    {
+        const float MAX_DISTANCE = 5000.0f;
+        float[] minDistanceFromKinect = { MAX_DISTANCE };
+        for (int i = 0; i < (int)trackerFrameData.NumOfBodies; i++)
+        {
+            var pelvisPosition = trackerFrameData.Bodies[i].JointPositions3D[(int)JointId.Pelvis];
+            Vector3 pelvisPos = new Vector3((float)pelvisPosition.X, (float)pelvisPosition.Y, (float)pelvisPosition.Z);
+            if (pelvisPos.magnitude < minDistanceFromKinect[0])
+            {
+                bodies[0] = i;
+                minDistanceFromKinect[0] = pelvisPos.magnitude;
+            }
+        }
```

Comment:
Consider writing this function using Linq query:
```
private IEnumerable<int> FindClosestTrackerBodies(BackgroundData trackerFrameData)
{
    const float MaxDistance = 5000.0f;
    return trackerFrameData.Bodies
        .Take(trackerFrameData.NumOfBodies)
        .Select((b, i) => new KeyValuePair<float, int>(b.JointPositions3D[(int)JointId.Pelvis)].Length(), i))
        .Where(p => p.Key < MaxDistance)
        .OrderBy(p => p.Key)
        .Select(p => p.Value);
}
```

Reply:
Hi I will change this to a query if that fits with coding standards, but I find the way it is written now easier to read and debug.